activiti-cloud-runtime-bundle-service/activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/runtime/MQServiceTaskIT.java

```diff
@@ -102,6 +109,55 @@ public void shouldRecoverFromFailure() {
             );
     }
 
+    @Test
+    public void shouldReplayRunningServiceTask() {
+        canFailConnector.reset();
+
+        //given
+        canFailConnector.setShouldSendResult(false);
+        Map<String, Object> variables = new HashMap<>();
+        variables.put("firstName", "John");
+        ProcessInstance procInst = runtimeService.startProcessInstanceByKey("MQServiceTaskErrorRecoverProcess",
+                                                                            "businessKey",
+                                                                            variables);
+        assertThat(procInst).isNotNull();
+        await("the service task should not send the result on the first try")
+            .untilTrue(canFailConnector.resultNotSent());
+
+        assertThat(taskService.createTaskQuery()
+                              .processInstanceId(procInst.getProcessInstanceId())
+                              .list()).isEmpty();
+        //when
+        IntegrationContext integrationContext = canFailConnector.getLatestReceivedIntegrationRequest()
+                                                                .getIntegrationContext();
+        canFailConnector.setShouldSendResult(true);
+        replayServiceTask(integrationContext);
+
+        await("the service task should send the result the second try")
+            .untilTrue(canFailConnector.resultSent());
+
+        //then
+        await("the execution should arrive in the human tasks which follows the service task")
+            .untilAsserted(() -> {
+                               List<Task> tasks = taskService.createTaskQuery().processInstanceId(procInst.getProcessInstanceId()).list();
+                               assertThat(tasks).isNotNull();
+                               assertThat(tasks).extracting(Task::getName).containsExactly("Schedule meeting after service");
+                           }
+            );
+
+        // and given
+        Task task = taskService.createTaskQuery()
+                               .processInstanceId(procInst.getProcessInstanceId())
+                               .singleResult();
+        // when
+        complete(task);
+
+        // then
+        assertThat(runtimeService.createProcessInstanceQuery()
+                                 .processInstanceId(procInst.getProcessInstanceId())
+                                 .list()).isEmpty();
```

Comment:
Could you, please, also add an assertion to check that no integration context related to this process instance was left behind?